File: leetcode/90_Subsets_II.java
class Solution {
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        Arrays.sort(nums);
        List<List<Integer>> result= new ArrayList<>();
        PossibleList(nums,result,new ArrayList<>(),0);
        return result;
    }
    public static void PossibleList(int []nums,List<List<Integer>>result, List<Integer>current, int start){
        result.add(new ArrayList<>(current));
        for(int i=start;i<nums.length;i++){
            if(i>start && nums[i]==nums[i-1]) {continue;}
            current.add(nums[i]);
            PossibleList(nums,result,current,i+1);
            current.remove(current.size()-1); 
        }
    }
}

File: leetcode/90_Subsets_II.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    set<vector<int>> res;
    vector<int> curr;
    void solve(vector<int>& nums, int strt){
        if(strt==nums.size()){
            res.insert(curr);
            return;
        }
        solve(nums, strt+1);
        curr.push_back(nums[strt]);
        solve(nums, strt+1);
        curr.pop_back();
    }

    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        solve(nums, 0);
        return vector<vector<int>>(res.begin(),res.end());
        
    }
};
int main(){
    Solution s;
    vector<int> nums = {1,2,2};
    vector<vector<int>> ans = s.subsetsWithDup(nums);
    for(auto v: ans){
        for(auto i: v){
            cout<<i<<" ";
        }
        cout<<endl;
    }
}